### PR TITLE
second task

### PR DESCRIPTION
**Pull request by Kate892F**

### CODE REVIEW

File: src/main/java/com/bookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+@Data
+public class CreateBookRequestDto {
+
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+
```
Comment (okuzan) [inline src/main/java/com/bookstore/dto/CreateBookRequestDto.java]:
```suggestion
```

File: src/main/java/com/bookstore/controller/imp/BookControllerImpl.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/books")
+@RequiredArgsConstructor
+public class BookControllerImpl implements BookController {
```
Comment (okuzan) [inline src/main/java/com/bookstore/controller/imp/BookControllerImpl.java]: it's an overkill, let's simplify by removing controller interface

**Review by okuzan — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 3]
```diff
@@ -56,6 +58,18 @@
...
+		<!-- https://mvnrepository.com/artifact/org.mapstruct/mapstruct -->
+		<dependency>
+			<groupId>org.mapstruct</groupId>
+			<artifactId>mapstruct</artifactId>
+			<version>${mapstruct.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+			<version>3.3.2</version>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
```

File: src/main/java/com/bookstore/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/books")
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

File: src/main/java/com/bookstore/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    private final BookService bookService;
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable(name = "id") Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/controller/BookController.java]:
```suggestion
    public BookDto getBookById(@PathVariable Long id) {
```

File: src/main/java/com/bookstore/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -48,4 +49,14 @@ public List<Book> findAll() {
...
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.find(Book.class, id);
+            return Optional.ofNullable(book);
+        } catch (Exception e) {
+            throw new DataProcessingException("Failed to get books", e);
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/repository/impl/BookRepositoryImpl.java]: wrong message

File: src/main/java/com/bookstore/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -12,14 +16,26 @@
...
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+            .map(bookMapper::toDto)
+            .toList();
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Failed to find book by id: " + id)
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/service/impl/BookServiceImpl.java]: It's better
```suggestion
                () -> new EntityNotFoundException("Cannot find book by id: " + id)
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
See comments

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**